include/seqan3/argument_parser/auxiliary.hpp

```diff
@@ -15,12 +15,210 @@
 #include <sstream>
 #include <vector>
 
+#include <seqan3/core/detail/customisation_point.hpp>
 #include <seqan3/core/detail/debug_stream_type.hpp>
+#include <seqan3/core/type_traits/basic.hpp>
 #include <seqan3/io/stream/concept.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/type_traits>
+
+namespace seqan3::custom
+{
+
+/*!\brief A type that can be specialised to provide customisation point implementations for the seqan3::argument_parser
+ *        such that third party types may be adapted.
+ * \tparam t The type you wish to specialise for.
+ * \ingroup argument_parser
+ *
+ * \details
+ *
+ * ### Named Enumerations
+ *
+ * In order to use a third party type within the seqan3::argument_parser::add_option or
+ * seqan3::argument_parser::add_positional_option call, you can specialise this struct in the following way:
+ *
+ * \include test/snippet/argument_parser/custom_argument_parsing_enumeration.cpp
+ *
+ * Please note that by default the `t const`, `t &` and `t const &` specialisations of this class inherit the
+ * specialisation for `t` so you usually only need to provide a specialisation for `t`.
+ *
+ * \note Only use this if you cannot provide respective functions in your namespace. See the tutorial
+ * \ref tutorial_argument_parser for an example of customising a type within your own namespace.
+ */
+template <typename t>
+struct argument_parsing
+{}; // forward
+
+//!\cond
+template <typename t>
+struct argument_parsing<t const> : argument_parsing<t>
+{};
+
+template <typename t>
+struct argument_parsing<t &> : argument_parsing<t>
+{};
+
+template <typename t>
+struct argument_parsing<t const &> : argument_parsing<t>
+{};
+//!\endcond
+
+} // seqan3::custom
+
+namespace seqan3::detail::adl_only
+{
+
+//!\brief Poison-pill overload to prevent non-ADL forms of unqualified lookup.
+template <typename t>
+std::unordered_map<std::string_view, t> enumeration_names(t) = delete;
+
+/*!\brief Functor definition for seqan3::enumeration_names.
+ *
+ * This struct is templetised because we need the original option_t next to s_option_t that may be wrapped
```

Comment:
```suggestion
 * This struct is templatised because we need the original option_t next to s_option_t that may be wrapped
```

Reply:
Note that I am not a big fan of the term "templatised struct" although I might have done this myself at certain points. It sounds very close to "template struct" or "template class" which is wrong. The correct term is "class template". → "We need a class template here, because ..."

This might sound very fussy, but language influences the way you think and many of the difficulties with templates come from a poor understanding. As authors of documentation we should pay attention to not teaching confusing names.